src/actions/remove.ts

```diff
@@ -0,0 +1,99 @@
+import { Async, ReaderT } from "crocks";
+import { __, assoc, compose, ifElse, isNil, path } from "ramda";
+import type { Config } from "../types.ts";
+
+interface H {
+  data: {
+    get: (id: string) => typeof Async;
+    remove: (id: string) => typeof Async;
+  };
+  cache: {
+    remove: (id: string) => typeof Async;
+  };
+  search: {
+    remove: (id: string) => typeof Async;
+  };
+}
+
+interface Context {
+  hyper: H;
+  id: string;
+  env?: Config;
+  doc?: unknown;
+}
+
+const { of, ask, lift } = ReaderT(Async);
+
+// first pass does not address any modes atomic vers non-atomic
+// it simply tries to delete everything if they exist or not
+/*
+const removeFromSearch = ({ hyper, id }: Context) =>
+  ask(() =>
+    hyper.search.remove(id)
+      .coalesce(() => ({ hyper, id }), () => ({ hyper, id }))
+  ).chain(lift);
+
+const removeFromCache = ({ hyper, id }: Context) =>
+  ask(() =>
+    hyper.cache.remove(id)
+      .coalesce(() => ({ hyper, id }), () => ({ hyper, id }))
+  ).chain(lift);
+*/
+const getFromData = (ctx: Context) =>
+  ctx.hyper.data.get(ctx.id).map(assoc("doc", __, ctx));
+const removeFromData = ({ hyper, id }: Context) => hyper.data.remove(id);
+const validateDocument = ifElse(
+  compose(isNil, path(["env", "schema"])),
+  Async.Resolved,
+  (ctx: Context) => {
+    // @ts-ignore safeParse invoke
+    const validate = Async.fromPromise(ctx.env.schema.spa.bind(ctx.env.schema));
```

Comment:
I'm assuming `spa` is an alias for safeParse on `zod` schemas? I think we should allow folks to validate with whatever they'd like. Perhaps we eventually make schema on environment an interface with a `parse` method, or something like that. Let me know thoughts. Regardless, doesn't have to be in this PR though.

Reply:
They can validate with whatever they like using hyper-connect, hyper-model is opinionated and will validate using `zod`, unless there is pushback. I think `zod` is emerging as the standard.